kbchat/kbchat_test.go

```diff
@@ -103,82 +103,290 @@ func deleteWorkingDir(workingDir string) error {
+	alice, err = Start(RunOptions{KeybaseLocation: kbLocation, HomeDir: dir, Oneshot: config.Bots["alice1"], StartService: true})
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in starting service: %v\n", err)
 	}
 
-	flag.Parse()
-	code := m.Run()
+	oneOnOneChannel = Channel{
+		Name: fmt.Sprintf("%s,%s", config.Bots["alice1"].Username, config.Bots["charlie1"].Username),
+	}
+	teamChannel = Channel{
+		Name:        config.Teams["acme"].Teamname,
+		Public:      false,
+		MembersType: "team",
+		TopicName:   config.Teams["acme"].Channel,
+		TopicType:   "chat",
+	}
 
-	err = kbc.Shutdown()
+	return alice, config, dir, oneOnOneChannel, teamChannel
+}
+
+func testTeardown(alice *API, dir string) {
+	err := alice.Shutdown()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error on service shutdown: %v\n", err)
 	}
 	err = deleteWorkingDir(dir)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error working directory teardown: %v\n", err)
 	}
-	os.Exit(code)
 }
 
 func TestGetUsername(t *testing.T) {
-	require.Equal(t, kbc.GetUsername(), config.Bots.Alice1.Username)
+	alice, config, dir, _, _ := testSetup()
+	require.Equal(t, alice.GetUsername(), config.Bots["alice1"].Username)
+	testTeardown(alice, dir)
 }
 
 func TestGetConversations(t *testing.T) {
-	conversations, err := kbc.GetConversations(false)
+	alice, _, dir, _, _ := testSetup()
+	conversations, err := alice.GetConversations(false)
 	require.NoError(t, err)
 	require.Greater(t, len(conversations), 0)
+	testTeardown(alice, dir)
 }
 
 func TestGetTextMessages(t *testing.T) {
-	channel := Channel{
-		Name: fmt.Sprintf("%s,%s", config.Bots.Alice1.Username, config.Bots.Charlie1.Username),
-	}
-	messages, err := kbc.GetTextMessages(channel, false)
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
 	require.NoError(t, err)
 	require.Greater(t, len(messages), 0)
+	testTeardown(alice, dir)
+}
+
+func TestSendMessage(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	text := "test SendMessage"
+
+	// Send the message
+	res, err := alice.SendMessage(oneOnOneChannel, text)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
+
+	// Read it to confirm it sent
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
 }
 
-func TestSendMessage(t *testing.T) {}
+func TestSendMessageByConvID(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	text := "test SendMessageByConvID"
 
-func TestSendMessageByConvID(t *testing.T) {}
+	// Retrieve conversation ID
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	convID := messages[0].ConversationID
```

Comment:
is this the only way to get the conversation id? seems weird.

Reply:
We could also get the list of all conversations and filter that by channel name to retrieve the conversation id. This seemed like a little less work. I don't think there's a great way to retrieve a conversation id directly from a provided api function.